### PR TITLE
ENH: OneDAL python ifaces for Linear Regression

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

File: onedal/linear_model/linear_model.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        const auto method = params["method"].cast<std::string>();
+        ONEDAL_PARAM_DISPATCH_VALUE(method, "norm_eq", ops, Float, method::norm_eq);
+        ONEDAL_PARAM_DISPATCH_VALUE(method, "by_default", ops, Float, method::norm_eq);
+        ONEDAL_PARAM_DISPATCH_THROW_INVALID_VALUE(method);
+    }
+
+    Ops ops;
+};
+
+auto get_onedal_result_options(const py::dict& params) {
```
Comment (samir-nasibli) [inline onedal/linear_model/linear_model.cpp:45]: dont use `onedal` in the func name, inconsistent

Comment (KulikovNikita) [reply onedal/linear_model/linear_model.cpp:45]: Nope. It is consistent. I.e. in [SVM](https://github.com/intel/scikit-learn-intelex/blob/e6495bbc43e5f84a147d29a54082567cc5901583/onedal/neighbors/neighbors.cpp#L103)

File: onedal/neighbors/neighbors.cpp [hunk 2]
```diff
@@ -24,13 +24,15 @@ namespace py = pybind11;
...
+namespace neighbors {
+
 template <typename Task, typename Ops>
 struct method2t {
     method2t(const Task& task, const Ops& ops) : ops(ops) {}
 
     template <typename Float>
     auto operator()(const py::dict& params) {
-        using namespace knn;
+        using namespace dal::knn;
```
[the remark this answers is not in the document]
Comment (KulikovNikita) [reply onedal/neighbors/neighbors.cpp:35]: It is impossible. The problem is that functions like `get_onedal_result_options` are present in multiple algorithms and linker can not resolve proper symbols. That's why I moved almost everything into the corresponding namespace and that's why this refactoring and correct namespace usage is needed.

File: sklearnex/linear_model/_common.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# ===============================================================================
+# Copyright 2021 Intel Corporation
```
[the remark this answers is not in the document]
Comment (KulikovNikita) [reply sklearnex/linear_model/_common.py]: Fixed

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/eda0cbdc-cc87-f12d-91f4-a4bf010d0e2e

**Comment by samir-nasibli** [comment]
> http://intel-ci.intel.com/eda0cbdc-cc87-f12d-91f4-a4bf010d0e2e

We have some related test fails.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Fix PEP8, CodeFactor and d4p/sklearnex building

**Pull request closed without merging**